pkg/quantile/agent.go

```diff
@@ -50,10 +54,13 @@ func (a *Agent) Reset() {
 }
 
 // Insert v into the sketch.
-func (a *Agent) Insert(v float64) {
-	a.Sketch.Basic.Insert(v)
+func (a *Agent) Insert(v float64, sampleRate float64) {
+	n := uint(math.Ceil(1 / sampleRate))
+	a.Sketch.Basic.InsertN(v, n)
 
-	a.Buf = append(a.Buf, agentConfig.key(v))
+	for i := uint(0); i < n; i++ {
+		a.Buf = append(a.Buf, agentConfig.key(v))
```

Comment:
Could this insert into the `CountBuf` and avoid having to insert n times?

Reply:
Yes but it'd be ugly, I'd need to reimplement flush, essentially (find the key, create a new one if it doesn't exist, call appendSafe ...). I think this is fine for now.

Reply:
I may be misunderstanding the `CountsBuf`, but isn't that what `flush()` already does?
https://github.com/DataDog/datadog-agent/blob/4c543ba9f7c7b67d7853f5644a9d930d92638313/pkg/quantile/agent.go#L44-L47

Reply:
oh yeah, I was thinking the countbuf was pre-interpolation but it's not, so that should work